Friendlier error messages when image path does not exist in cv2.seamlessClone()

##### System information (version)

- OpenCV => 4.2
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio Code 1.56.2

##### Detailed description
When running cv2.seamlessClone() the error is a bit misleading when the incorrect image path is supplied. It doesn't make it obvious that the problem is in the path

##### Steps to reproduce

```
import cv2
# Read images
DATA_PATH = "data_set/data"
source = cv2.imread(f"{DATA_PATH}/this image doesn't exist.png")
target = cv2.imread(f"{DATA_PATH}/10226_bg_27794.png")
swap = cv2.imread(f"{DATA_PATH}/10226_sw_27794_19200.png")
mask = cv2.imread(f"{DATA_PATH}/10226_mask_27794_19200.png")
# mask = mask/255
center = (112, 112)
output_normal = cv2.seamlessClone(swap, target, mask, center, cv2.NORMAL_CLONE)
output_mixed = cv2.seamlessClone(swap, target, mask, center, cv2.MIXED_CLONE)

# Output
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-3-faf121550370> in <module>
     10 # mask = mask/255
     11 center = (112, 112)
---> 12 output_normal = cv2.seamlessClone(swap, target, mask, center, cv2.NORMAL_CLONE)
     13 output_mixed = cv2.seamlessClone(swap, target, mask, center, cv2.MIXED_CLONE)

error: OpenCV(4.5.1) C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-kh7iq4w7\opencv\modules\core\src\matrix.cpp:811: error: (-215:Assertion failed) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function 'cv::Mat::Mat'
```


##### Issue submission checklist

 - [x] I report the issue, it's not a question
   <!--
   OpenCV team works with forum.opencv.org, Stack Overflow and other communities
   to discuss problems. Tickets with question without real issue statement will be
   closed.
   -->
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   <!--
   Places to check:
   * OpenCV documentation: https://docs.opencv.org
   * FAQ page: https://github.com/opencv/opencv/wiki/FAQ
   * OpenCV forum: https://forum.opencv.org
   * OpenCV issue tracker: https://github.com/opencv/opencv/issues?q=is%3Aissue
   * Stack Overflow branch: https://stackoverflow.com/questions/tagged/opencv
   -->
 - [x] I updated to latest OpenCV version and the issue is still there
   <!--
   master branch for OpenCV 4.x and 3.4 branch for OpenCV 3.x releases.
   OpenCV team supports only latest release for each branch.
   The ticket is closed, if the problem is not reproduced with modern version.
   -->
 - [x] There is reproducer code and related data files: videos, images, onnx, etc
   <!--
   The best reproducer -- test case for OpenCV that we can add to the library.
   Recommendations for media files and binary files:
   * Try to reproduce the issue with images and videos in opencv_extra repository
     to reduce attachment size
   * Use PNG for images, if you report some CV related bug, but not image reader
     issue
   * Attach the image as archive to the ticket, if you report some reader issue.
     Image hosting services compress images and it breaks the repro code.
   * Provide ONNX file for some public model or ONNX file with with random weights,
     if you report ONNX parsing or handling issue. Architecture details diagram
     from netron tool can be very useful too. See https://lutzroeder.github.io/netron/
   -->

--- a/modules/imgcodecs/src/loadsave.cpp
+++ b/modules/imgcodecs/src/loadsave.cpp
@@ -226,8 +226,10 @@ static ImageDecoder findDecoder( const String& filename ) {
     FILE* f= fopen( filename.c_str(), "rb" );
 
     /// in the event of a failure, return an empty image decoder
-    if( !f )
+    if( !f ) {
+        CV_LOG_WARNING(NULL, "imread_('" << filename << "'): can't open/read file: check file path/integrity");
         return ImageDecoder();
+    }
 
     // read the file signature
     String signature(maxlen, ' ');

--- a/modules/photo/src/seamless_cloning.cpp
+++ b/modules/photo/src/seamless_cloning.cpp
@@ -67,6 +67,7 @@ static Mat checkMask(InputArray _mask, Size size)
 void cv::seamlessClone(InputArray _src, InputArray _dst, InputArray _mask, Point p, OutputArray _blend, int flags)
 {
     CV_INSTRUMENT_REGION();
+    CV_Assert(!_src.empty());
 
     const Mat src  = _src.getMat();
     const Mat dest = _dst.getMat();

